apps/emqx_connector/src/emqx_connector.erl

```diff
@@ -458,3 +460,55 @@ ensure_no_channels(Configs) ->
         {error, Reason, _State} ->
             {error, Reason}
     end.
+
+to_bin(A) when is_atom(A) -> atom_to_binary(A, utf8);
+to_bin(B) when is_binary(B) -> B.
+
+validate_connector_name(ConnectorName0) ->
+    ConnectorName = to_bin(ConnectorName0),
+    case re:run(ConnectorName, ?MAP_KEY_RE, [{capture, none}]) of
```

Comment:
there is `emqx_resource:validate_name/1`

Reply:
:heavy_check_mark: